src/Swarm/TUI/Controller.hs

```diff
@@ -515,13 +515,44 @@ handleREPLEvent s (ControlKey 'c') =
       & gameState . robotMap . ix 0 . machine %~ cancel
 handleREPLEvent s (Key V.KEnter) =
   if not $ s ^. gameState . replWorking
-    then case processTerm' topTypeCtx topCapCtx entry of
-      Right mt -> do
-        let s' = advanceREPL s
-            s'' = maybe id startBaseProgram mt s'
-        continue s''
-      Left err ->
-        continue $ s & uiState . uiError ?~ err
+    then case entry of
```

Comment:
nit, perhaps the case expression could return the state transformation so that we can factor out the continue like this:

```suggestion
    then continue $ s & case entry of
```

Reply:
Absolutely! will commit along with other changes